integration-tests/cypress/commands/api/device.ts

```diff
@@ -0,0 +1,539 @@
+  (oldName: string, newName: string, newGroup: string, password: string = null, keepName:boolean = false, statusCode: number = 200) => {
+    var uniqueName;
+    logTestDescription(
+      `Reregister camera '${newName}' in group '${newGroup}'`,
+      {
+        camera: newName,
+        group: newGroup
+      },
+      true
+    );
+
+    if(password==null) {
+      password="p"+getTestName(newName);
+    };
+
+    if(keepName==false) {
+      uniqueName=getTestName(newName);
+    } else {
+      uniqueName=newName;
+    };
+
+    const data = {
+      newName: uniqueName,
+      newPassword: password,
+      newGroup: getTestName(newGroup)
+    };
+    const fullUrl = v1ApiPath('devices/reregister');
+
+    makeAuthorizedRequestWithStatus(
+      {
+        method: "POST",
+        url: fullUrl,
+        body: data
+      },
+      oldName,
+      statusCode
+    ).then((response) => {
+      const id = response.body.id;
+      saveCreds(response, newName, id);
+    });
+  }
+);
+
+
+Cypress.Commands.add(
+  "apiShouldFailToCreateCamera",
+  (
+    cameraName: string,
+    group: string,
+    password: string = null,
+    makeCameraATestName = true,
+    log = true
+  ) => {
+    logTestDescription(
+      `Check that user cannot create camera '${cameraName}' in group '${group} '`,
+      {
+        camera: cameraName,
+        group: group
+      },
+      log
+    );
+
+    const request = createCameraDetails(cameraName, group, password, null, makeCameraATestName);
+    checkRequestFails(request);
+  }
+);
+
+function createCameraDetails(
+  cameraName: string,
+  group: string,
+  password: string,
+  saltId: number,
+  makeCameraNameTestName = true
+): any {
+  const fullName = makeCameraNameTestName
+    ? getTestName(cameraName)
+    : cameraName;
+
+  if(password==null) {
+    password = "p" + fullName;
+  };
+
+  const data = {
+    devicename: fullName,
+    password: password,
+    group: getTestName(group)
+  };
+
+  if(saltId!=null) {
+    data.saltId=saltId;
+  };
+
+  return {
+    method: "POST",
+    url: v1ApiPath("devices"),
+    body: data
+  };
+}
+
+
+Cypress.Commands.add("apiCheckDevices", (userName: string, expectedDevices: [ComparableDevice], params: string = {},  statusCode: number = 200) => {
+  const fullUrl = v1ApiPath('devices',params);
+
+  logTestDescription(
+      `${userName} Check devices seen by  user '${userName}'`,
+      { user: userName},
+      true
+  );
+
+  makeAuthorizedRequestWithStatus(
+      {
+        method: "GET",
+        url: fullUrl,
+        body: null
+      },
+      userName,
+      statusCode
+    ).then((response)=>{
+       if(statusCode==null || statusCode==200) {
+	  var devices=response.body.devices.rows;
+	  //TODO: Issue 63.  Reenable this when devices count is correct
+          //expect(response.body.devices.count).to.equal(expectedDevices.length);
+          expect(devices.length).to.equal(expectedDevices.length);
+	  var dev_count;
+	  var user_count;
```

Comment:
unsued

Reply:
this block also needs to be indented properly

Reply:
where applicabe use let instead of var

Reply:
done